homeassistant/config_entries.py

```diff
@@ -1528,6 +1531,99 @@ class OptionsFlow(data_entry_flow.FlowHandler):
     handler: str
 
 
+class HelperCommonFlowHandler:
+    """Handle a config or options flow for helper."""
+
+    def __init__(
+        self,
+        handler: HelperConfigFlowHandler | HelperOptionsFlowHandler,
+        config_entry: ConfigEntry | None,
+    ) -> None:
+        """Initialize a common handler."""
+        self._handler = handler
+        self._options = dict(config_entry.data) if config_entry is not None else {}
+
+    @staticmethod
+    async def validate_input(
+        hass: HomeAssistant, data: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Validate the user input allows us to connect."""
+
+        # TODO: Some fancy validation of the selectors?
```

Comment:
This could be stuff like making sure an entity_id really exists, etc.

Reply:
if we do that, we should add a new method to selectors to `validate_with_hass` or something.

Reply:
OK, let's consider that in a follow-up PR then.